sdk/Sdk.Generators/FunctionExecutor/FunctionExecutorGenerator.cs

```diff
@@ -38,9 +38,9 @@ public void Execute(GeneratorExecutionContext context)
 
             var entryAssemblyFuncs = GetSymbolsMethodSyntaxes(receiver.CandidateMethods, context);
             var dependentFuncs = GetDependentAssemblyFunctionsSymbols(context);
-            var allMethods = entryAssemblyFuncs.Concat(dependentFuncs);
+            var allMethods = entryAssemblyFuncs.Concat(dependentFuncs).ToArray();
```

Comment:
Happy to revert if you disklike it. Based on my short analysis I was thinking the Any doesn't have a materialized enumerable so it can use TryGetNonEnumeratingCount underneath the covers and then Any will enumerate the whole collection away so will the parser. Therefore I concluded it is better to materialize once.

Reply:
`Any()` won't enumerate the _entire_ collection. It will get the first result and then return.

Reply:
I think this PR has edit rights if you want to revert that commit before merging. Otherwise I can do it myself some time tomorrow and force push.

Reply:
I ran a quick benchmark and `ToArray`is indeed slower and more allocation heavy. Thanks for catching my mistake here and sorry for the derail. I will in the next contribution pay more attention to those details and not assume (which I'm normally better at but as I said I might have done this change a bit too quick without thinking further, aka Lazy daniel)